### PR TITLE
[GeoMechanicsApplication] Formatted the custom_workflow folder using clang-format

### LINKED ISSUE #11919: [GeoMechanicsApplication] Format GeoMechanics code base using clang-format
In this issue, the geomechanics code base will be formatted using clang-format. It will be done folder by folder, which is why there are multiple PRs linked to this issue.

### PR DESCRIPTION
**Pull request by rfaasse**

### CODE REVIEW

File: applications/GeoMechanicsApplication/.clang-format [hunk 1]
```diff
@@ -36,6 +36,7 @@ PenaltyBreakFirstLessLess: 300
 PenaltyExcessCharacter: 1
 PenaltyReturnTypeOnItsOwnLine: 1000
 PointerAlignment: Left
+SeparateDefinitionBlocks: Always
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/.clang-format:40]: Useful addition

File: applications/GeoMechanicsApplication/custom_workflows/adaptive_time_incrementor.cpp [hunk 2]
```diff
@@ -16,70 +16,67 @@
...
-    mEndTime(EndTime),
-    mDeltaTime(std::min(StartIncrement, EndTime - StartTime)), // avoid exceeding the end time
-    mMaxNumOfCycles(MaxNumOfCycles),
-    mReductionFactor(ReductionFactor),
-    mIncreaseFactor(IncreaseFactor),
-    mMaxDeltaTime(MaxTimeStepFactor*mDeltaTime),
-    mMinNumOfIterations(MinNumOfIterations),
-    mMaxNumOfIterations(MaxNumOfIterations)
+                                                 std::size_t MaxNumOfIterations)
+    : TimeIncrementor(),
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/custom_workflows/adaptive_time_incrementor.cpp:31]: Good location for :

File: applications/GeoMechanicsApplication/custom_workflows/adaptive_time_incrementor.cpp [hunk 3]
```diff
@@ -16,70 +16,67 @@
...
-                                                                << mMaxNumOfIterations << ")";
-    KRATOS_ERROR_IF(mReductionFactor  >  1.0) << "Reduction factor must not be greater than 1, but got " << mReductionFactor;
-    KRATOS_ERROR_IF(mReductionFactor  <= 0.0) << "Reduction factor must be positive, but got " << mReductionFactor;
-    KRATOS_ERROR_IF(mIncreaseFactor   <  1.0) << "Increase factor must be greater than or equal to 1, but got " << mIncreaseFactor;
-    KRATOS_ERROR_IF(MaxTimeStepFactor <  1.0) << "Max_delta_time_factor must be greater than or equal to 1, but got " << MaxTimeStepFactor;
+    KRATOS_ERROR_IF(mMaxNumOfCycles == std::size_t(0))
+        << "Maximum number of cycles must be positive";
+    KRATOS_ERROR_IF(mMinNumOfIterations >= mMaxNumOfIterations)
+        << "Minimum number of iterations (" << mMinNumOfIterations
+        << ") is not less than maximum number of iterations (" << mMaxNumOfIterations << ")";
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/custom_workflows/adaptive_time_incrementor.cpp:48]: Good changes to short lines

File: applications/GeoMechanicsApplication/custom_workflows/time_loop_executor.hpp [hunk 4]
```diff
@@ -119,11 +117,11 @@ public :
         }
     }
 
-    std::unique_ptr<TimeIncrementor>  mTimeIncrementor;
-    std::function<bool()>             mCancelDelegate;
-    std::function<void(double)>       mProgressDelegate;
+    std::unique_ptr<TimeIncrementor> mTimeIncrementor;
+    std::function<bool()> mCancelDelegate;
+    std::function<void(double)> mProgressDelegate;
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/custom_workflows/time_loop_executor.hpp]: It seems the old format is better. clang has 'bool Enabled'. Can we use it?

**Review by markelov208 — COMMENTED** [review]
Richard, very good, for example, the number of lines is reduced! Just one comment on bool Enabled.

File: applications/GeoMechanicsApplication/custom_workflows/time_loop_executor.hpp (hunk 4 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/custom_workflows/time_loop_executor.hpp]: Good suggestion, I missed the AlignConsecutiveDeclarations option when checking the documentation, so I didn't think it was possible!

**Review by markelov208 — APPROVED** [review]
Richard, many thanks for the clang format improvements.

**Pull request merged**